source/slang/emit.cpp

```diff
@@ -76,6 +78,21 @@ void requireGLSLVersionImpl(
     }
 }
 
+void requireGLSLHalfExtension(ExtensionUsageTracker* tracker)
+{
+    if (!tracker->hasHalfExtension)
```

Comment:
Why add this condition? The `ExtensionUsageTracker` should already have logic to not add the same extension more than once.

Reply:
It's not totally necessary - and was originally there when I was adding 4 extensions, and it seemed kind of foolish to do 4 set lookups for every time 'half' was seen. Now with only 2 extensions perhaps it's not required.